src/h5web/dataset-visualizer/DimensionMapper.tsx

```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+import ReactSlider from 'react-slider';
+import { Vis, DimensionMapping } from './models';
+import { useVisProps } from './VisProvider';
+import ButtonGroup from '../visualizations/shared/ButtonGroup';
+import styles from './DimensionMapper.module.css';
+
+interface Props {
+  activeVis: Vis;
+  setMapping: (d: DimensionMapping) => void;
+}
+
+function DimensionMapper(props: Props): JSX.Element {
+  const { activeVis, setMapping } = props;
+  const { rawDims, mapping } = useVisProps();
+
+  if (activeVis === Vis.Line && rawDims.length === 2) {
+    return (
+      <div className={styles.mapper}>
+        <div className={styles.buttonGroupWrapper}>
+          <span className={styles.buttonGroupLabel}>X</span>
+          <ButtonGroup
+            className={styles.buttonGroup}
+            buttonClassName={styles.btn}
+            buttons={Object.keys(rawDims).map(dimKey => {
```

Comment:
Even though there are only 2 dims, I tried to keep a syntax that can be extensible to higher dimensions.